packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -96,7 +96,9 @@ export const DataSourceHttpSettings: React.FC<HttpSettingsProps> = (props) => {
       width={20}
       options={ACCESS_OPTIONS}
       value={ACCESS_OPTIONS.filter((o) => o.value === dataSourceConfig.access)[0] || DEFAULT_ACCESS_OPTION}
-      onChange={(selectedValue) => onSettingsChange({ access: selectedValue.value })}
+      onChange={(selectedValue) => {
+        selectedValue && onSettingsChange({ access: selectedValue.value });
```

Comment:
Not sure this is the best approach, we're basically skipping on change if the value is falsy, so resetting the value will be not possible. It should be smth like this at least: `onSettingsChange({ access: selectedValue?.value });`.  

In general it'd be best if the null value was handled in the `onChangeWithEmpty` callback, e.g. by returning ` return onChange({ label: undefined, value: undefined })` if the value is null.

Still I think that a lot of code depends on the previous types (not only grafana but plugins, etc) so introducing a breaking change like this might not be a good idea.

Reply:
Just to be clear - there's no actual change in functionality of the onChange function here. onChange has always passed through null, just when `isClearable` is set and the cross button is clicked to clear it. This PR does not cause null to be passed through more (setting value from outside doesn't trigger onChange).

I don't think these changes to components using Select change anything in practice, as null is only a possible value if isClearable is set, and none of these components use that afaict. If they are, they are currently throwing at runtime.

I'm less confident about changing the functionality of onChange to send through `onChange({ label: undefined, value: undefined })` when the value is cleared, as that's a breaking change in the functionality 

The onChange type change isn't really all that related to my changes in this PR - maybe its better if I [undo adding `| null` to `SelectValue`](https://github.com/grafana/grafana/pull/32136/files#diff-b93a6d3d1c14917c7d1d0dbcfba3f31bc1901b913d6888b8ab79fb3d85da0b7eR4) and and just make the type of `value` by `SelectValue<T> | null`? Thoughts? That would allow the seperate concern of the incorrect onChange type to be dealt with specificly, rather than incidently in this PR.

Reply:
> I don't think these changes to components using Select change anything in practice, as null is only a possible value if isClearable is set, and none of these components use that afaict. If they are, they are currently throwing at runtime.

My main concern was that we now need to handle null values in `onChange` callback. 

> I'm less confident about changing the functionality of onChange to send through onChange({ label: undefined, value: undefined }) when the value is cleared, as that's a breaking change in the functionality

Good point! Forgot about clearing functionality.